R/meltAssay.R

```diff
@@ -0,0 +1,135 @@
+#' @title meltAssay
+#'
+#' Converts a \code{\link[=MicrobiomeExperiment-class]{MicrobiomeExperiment}} object into a
+#' long data.frame which can be used for \code{\link[tidyverse]{tidyverse}}-tools.
+#'
+#' @param x a numeric matrix or a \code{\link[=MicrobiomeExperiment-class]{MicrobiomeExperiment}}
+#'          object containing a tree.
+#' @param abund_values Default is counts, must be one of "counts", "relabundance", "both".
+#' @param colData_choice Choice of colData columns to add. By default it will consider `"all"`
+#' columns. Specific columns can be provided as `c("col1", "col2", ...)`. If `colData_choice=NA`
+#' only the specified `assayName` will be converted to long data.frame
+#' @param ... optional arguments not used.
+#' @return A data.frame.
+#' @importFrom SummarizedExperiment assayNames rowData rowData<-
+#' @importFrom tibble rownames_to_column
+#' @importFrom tidyr pivot_longer
+#' @importFrom dplyr left_join mutate_if %>%
+#' @export
+setGeneric("meltAssay",
+  signature = "x",
+  function(x, colData_choice = "all", abund_values = "counts", ...) {
+    standardGeneric("meltAssay")
+  }
+)
+
+
+#' @rdname meltAssay
+#' @aliases meltAssay
+#'
+#' @importFrom SummarizedExperiment rowData rowData<- colData colData<-
+#'
+#' @export
+setMethod("meltAssay",
+  signature = c(x = "MicrobiomeExperiment"),
+  function(x, colData_choice = "all", abund_values = "counts") {
+    # input check
+    if (isFALSE(any(abund_values == c("counts", "relabundance", "both")))) {
```

Comment:
You can checkout the function `.check_abund_values` in `MicrobiomeExperiment`. I added that as the universal check for a certain assay to be present. I also don't like to hard code assay names. How the user names their assay and maybe wants to compare two assays of the same type, is up to them.